source/server.go

```diff
@@ -53,6 +55,11 @@ func (s *Server) Stream(stream pb.SourceService_StreamServer) error {
 		sdk.Logger(s.openContext).Warn().Msg("only one client connection is supported")
 		return errors.New("only one client connection is supported")
 	}
+	// empty channel
+	for len(s.StreamIsSetCh) > 0 {
+		<-s.StreamIsSetCh
+	}
+	s.StreamIsSetCh <- true
```

Comment:
So we are emptying the channel just to put a value in it? 🤔 We can simplify:
```
select {
	case s.StreamIsSetCh <- true:
		// we successfully wrote to the channel
	default:
		// channel already contains a value
}
```

Reply:
I had a buffered channel of more than one record in the beginning 😅

Reply:
@lovromazgon I'm emptying it to make sure that there's always only one value inside the channel or none, if we made this change, then when a client disconnected and reconnected more than one time without the channel getting emptied, the code will block here.

Reply:
oh sorry had something else on my mind 😅